src/mem/cache/prefetch/stride.hh

```diff
@@ -137,7 +137,7 @@ class Stride : public Queued
         SatCounter8 confidence;
     };
     typedef AssociativeSet<StrideEntry> PCTable;
-    std::unordered_map<int, PCTable> pcTables;
+    std::unordered_map<int, PCTable*> pcTables;
```

Comment:
This explains why new is now needed. Why was this change to pointer needed?

Reply:
This change should be reverted

Reply:
Bump

Reply:
Maybe it is because the copy and assignment operators were removed from AssociativeCache (just a guess). Nonetheless, I think this is a bug that is introduced by this PR, and that it needs to be fixed in order for this PR to be merged.

Reply:
I think using a pointer is actually better than copying the entire array elements for this case. Can you please check this updated code where we are saving a pointer to the table instead of copying it?

Reply:
There was only one place initializing the table:
`pcTables.insert(std::make_pair(context, PCTable(`

After the table is constructed, it is not deleted nor overwritten, and there are possibly one or two static copies (if, and only if, make_pair or insert trigger one). However, with pointers we end up with a frequent dereference overhead. I would say that frequent overhead is likely worse than a single one upon initialization.

Reply:
In any case, when a small side change generates a discussion that makes the review longer, if possible IMHO it is better to drop it from the main PR and make a proposal PR just for the side change in order to get the main PR going. I believe this is my last remaining request, but I will double check over the weekend.

Reply:
> There was only one place initializing the table: `pcTables.insert(std::make_pair(context, PCTable(`
> 
> After the table is constructed, it is not deleted nor overwritten, and there are possibly one or two static copies (if, and only if, make_pair or insert trigger one). However, with pointers we end up with a frequent dereference overhead. I would say that frequent overhead is likely worse than a single one upon initialization.

I don't quite agree with the overhead statement here. Saving a pointer and referencing the newly constructed table in the map is as efficient as referencing an object within the container. And we actually do want a pointer as you can see in the findTable() function, it returns a pointer eventually.

But that discussion is not that relevant for the current patch, so I reverted it to what it was previously and am storing the PCTable object in the map using piecewise constructor so that we don't have to copy the object immediately after creating it.